src/pal/src/CMakeLists.txt

```diff
@@ -161,16 +187,29 @@ set(SOURCES
 add_library(coreclrpal
   STATIC
   ${SOURCES}
+  ${ARCH_SOURCES}
   ${PLATFORM_SOURCES}
 )
 
 if(CMAKE_SYSTEM_NAME STREQUAL Linux)
-  target_link_libraries(coreclrpal
-    pthread
-    rt
-    dl
-    unwind
-  )
+  # On ARM linking with libunwind will break C++ exceptions unless we first
+  # link with gcc_s, this is a libunwind issue
+  if(PAL_CMAKE_PLATFORM_ARCH_ARM)
+    target_link_libraries(coreclrpal
+      gcc_s
```

Comment:
@janvorli Yes, basically C++ exceptions when trying to call the _Unwind_* functions, end up using the ones in libunwind instead of libgcc_s, which evidently are not binary compatible with one another.  I've seen this on another platform, and wanted to discuss it with you at some point (we may want to fork llvm libunwind without the C++ public API or something), but thats a topic for another issue/pr.